src/twister2/log_parser/log_parser_abstract.py

```diff
@@ -7,10 +7,14 @@
 
 
 class LogParserAbstract(abc.ABC):
+    class State(str, enum.Enum):
```

Comment:
Yes, I agree with you and I don't like this solution - but I don't have idea how to do it better. I wanted to emphasis somehow that this `State` describe only `LogParser` state to avoid misunderstanding with ZTest statuses and pytest internal test statuses. I don't see any sense to extract this internal `State` class as a standalone class when it is used only with `LogParser` - with this approach I need to import this class almost every time when I would like to use `LogParser`. To simplify it I can just replace this `State` class by three string constants, like:
```python
STATE_UNKNOWN = "UNKNOWN"
STATE_FAILED = "FAILED"
STATE_PASSED = "PASSED"
```
But with this approach I'll lost possibility to prompt potential user what kind of states are available (what is cleaner when I have Enum class with limited options and I can use it as a type of `self.state` attribute).

Reply:
What about having this `State` class implemented outside the `LogParserAbstract` class and just have added "constant" to `LogParserAbstract` class like:
```python
class LogParserAbstract:
    STATE = State
```
Also referring to that constant in the code looks better, in my opinion, when we use `parser.STATE` instead of `parser.State`. Second option looks like you have some variable with a name starting from capital latter.